air/src/utils.rs

```diff
@@ -0,0 +1,216 @@
+//! A collection of utility functions helpful in defining AIR's.
+
+use core::array;
+
+use p3_field::{AbstractField, Field};
+
+use crate::AirBuilder;
+
+/// Pack a collection of bits into a number.
+///
+/// Given vec = [v0, v1, ..., v_n] returns v0 + 2v_1 + ... + 2^n v_n
+pub fn two_pack<AF, Var, I>(iter: I) -> AF
+where
+    AF: AbstractField,
+    Var: Into<AF> + Clone,
+    I: DoubleEndedIterator<Item = Var>,
+{
+    let mut output = AF::ZERO;
+    for elem in iter.rev() {
+        output = output.double();
+        output += elem.clone().into();
+    }
+    output
+}
+
+/// Computes the arithmetic generalization boolean `xor`.
+///
+/// For boolean inputs, `x ^ y = x + y - 2 xy`.
+pub fn xor<AF: AbstractField>(x: AF, y: AF) -> AF {
```

Comment:
nit: since there will be many calls to these, I would tag them with either `#[inline]` or `#[inline(always)]`. Especially now that they're in a separate crate, so I believe LLVM won't otherwise consider inlining if LTO is off.